### PR TITLE
[#1925] Add delete-track-events dialog.

### PR DESCRIPTION
**Pull request by cadecairos**
Lighthouse Ticket: https://webmademovies.lighthouseapp.com/projects/65733-popcorn-maker/tickets/1925

### CODE REVIEW

File: src/dialog/dialogs/delete-track-events.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!--  This Source Code Form is subject to the terms of the MIT license
+      If a copy of the MIT license was not distributed with this file, you can
+      obtain one at https://raw.github.com/mozilla/butter/master/LICENSE -->
+
+<div class="butter-dialog butter-form small">
+  <p>Would you like to delete all track events?</p>
```
Comment (mjschranz) [inline src/dialog/dialogs/delete-track-events.html]: I think a more appropriate message might be along the lines of `Would you like all track events to be removed with the media change as well?`

That might not be that great but just something to distinguish this from our normal trackevent deletion dialog.

File: src/editor/media-editor.js [hunk 2]
```diff
@@ -194,11 +194,27 @@ define( [ "util/lang", "util/uri", "util/keys", "editor/editor", "text!layouts/m
...
-  function onMediaReady() {
+  function finishMediaLoad() {
     showError( false );
     setLoadSpinner( false );
   }
 
+  function onMediaReady() {
+    var dialog;
+
+    if ( Butter.app.currentMedia.hasTrackEvents() ) {
```
Comment (mjschranz) [inline src/editor/media-editor.js]: You could just store the butter instance you receive on line 218.

File: src/editor/media-editor.js [hunk 3]
```diff
@@ -194,7 +201,20 @@ define( [ "util/lang", "util/uri", "util/keys", "editor/editor", "text!layouts/m
...
+    var dialog = Dialog.spawn( "delete-track-events", {
+      data: _butter
+    });
+
+    dialog.open();
+  }
+
   function onMediaReady() {
+    if ( _clearEventsOnReady ) {
+      spawnClearDialog();
```
Comment (ScottDowne) [inline src/editor/media-editor.js]: Why make this a function? Why not just put the code in this block?

File: src/editor/media-editor.js [hunk 4]
```diff
@@ -194,7 +201,20 @@ define( [ "util/lang", "util/uri", "util/keys", "editor/editor", "text!layouts/m
     showError( true );
   }
 
+  function spawnClearDialog() {
+    _clearEventsOnReady = false;
```
Comment (ScottDowne) [inline src/editor/media-editor.js]: Why set this to false?

Is the checkbox deselected? If not, is there a reason for this?

You can make it so this value only gets changed when the checkbox is clicked.

File: src/editor/media-editor.js [hunk 5]
```diff
@@ -194,7 +201,20 @@ define( [ "util/lang", "util/uri", "util/keys", "editor/editor", "text!layouts/m
...
+
+    var dialog = Dialog.spawn( "delete-track-events", {
+      data: _butter
+    });
+
+    dialog.open();
+  }
+
   function onMediaReady() {
+    if ( _clearEventsOnReady ) {
```
Comment (ScottDowne) [inline src/editor/media-editor.js]: Also check that we still have track events here.

**Pull request merged**